code/modules/projectiles/pins.dm

```diff
@@ -249,28 +257,30 @@
 /obj/item/firing_pin/paywall/pin_auth(mob/living/user)
 	if(!istype(user))//nice try commie
 		return FALSE
-	if(!pin_owner)
-		to_chat(user, "<span class='warning'>ERROR: Installed firing pin has no active bank account linked to it!</span>")
-		return FALSE
 	if(ishuman(user))
 		var/datum/bank_account/credit_card_details
 		var/mob/living/carbon/human/H = user
-		if(H in gun_owners)
-			return TRUE
 		if(H.get_bank_account())
 			credit_card_details = H.get_bank_account()
+		if(H in gun_owners)
+			if(multi_payment && credit_card_details)
+				if(credit_card_details.adjust_money(-payment_amount))
+					pin_owner.registered_account.adjust_money(payment_amount)
+					return TRUE
+				to_chat(user, "<span class='warning'>ERROR: User balance insufficent for successful transaction!</span>")	
+				return FALSE	
+			return TRUE				
 		if(credit_card_details && !active_prompt)
-			var/license_request = alert(usr, "Do you wish to pay [payment_amount] for license of this [gun.name]?", "License Purchase", "Yes", "No")
+			var/license_request = alert(usr, "Do you wish to pay [payment_amount] credit[( payment_amount > 1 ) ? "s" : ""] for [( multi_payment ) ? "each shot of [gun.name]" : "usage license of [gun.name]"]?", "Weapon Purchase", "Yes", "No")
```

Comment:
im skimming here but does this go off every time you try to shoot? that's funny if so but also what happens if you keep this prompt open, throw the guy away somewhere, then accept the prompt while not being near the gun?

Reply:
There's a check higher up that looks to see if you've already bought the permit so you will only see the prompt once

I should probably add a range check for that oops